### PR TITLE
MRG: Fix surf ori

### PR DESCRIPTION
**Pull request by Eric89GXL**
No fix yet, just currently adds code to demonstrate the error referred to by #3140.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Shortcut to failing build:

https://travis-ci.org/mne-tools/mne-python/jobs/124075056

File: doc/whats_new.rst [hunk 1]
```diff
@@ -109,6 +109,8 @@ BUG
 
     - Fix bug in colormap selection in :func:`mne.Evoked.plot_projs_topomap` by `Jaakko Leppakangas`_
 
+    - Fix bug in source normal adjustment that occurred when 1) patch information is available (e.g., when distances have been calculated) and 2) points are excluded from the source space (by inner skull distance) by `Eric Larson`_
```
Comment (Eric89GXL) [inline doc/whats_new.rst]: This one is a bit bad -- our surface normals were broken for probably many forward solutions :(

File: mne/source_space.py [hunk 2]
```diff
@@ -2030,6 +2038,9 @@ def _filter_source_spaces(surf, limit, mri_head_t, src, n_jobs=1,
             extras += [limit]
             logger.info('%d source space point%s omitted because of the '
                         '%6.1f-mm distance limit.' % tuple(extras))
+        # Adjust the patch inds as well if necessary
+        if s.get('patch_inds') is not None:
+            s['patch_inds'] = s['patch_inds'][~outside]
```
Comment (Eric89GXL) [inline mne/source_space.py]: This is the fix -- the `patch_inds` are used to adjust the surface normals, and if points are omitted/filtered out during the forward solution calculation stage (by distance from or being outside the inner skull surface) then these would be off by some indices.

**Comment by Eric89GXL** [comment]
Ready for review/merge from my end.

**Comment by agramfort** [comment]
arff :(

let's merge when CIs are happy

**Comment by Eric89GXL** [comment]
They were on the previous commit, and I just re-did the fix with a cleaner solution to call our existing `_add_patch_info(s)` function instead of doing the manual re-indexing. Should be more robust.

Will merge when they come back happy.

**Pull request merged**